features/config/rest/impl/src/main/java/org/opennms/features/config/rest/impl/ConfigSwaggerConverter.java

```diff
@@ -0,0 +1,388 @@
+import java.util.regex.Pattern;
+
+public class ConfigSwaggerConverter {
+    private final Logger LOG = LoggerFactory.getLogger(ConfigSwaggerConverter.class);
+
+    private final Map<ConfigItem, Schema<?>> schemasByItem = new LinkedHashMap<>();
+    private final Map<ConfigItem, String> pathsByItem = new LinkedHashMap<>();
+    private final Map<String, PathItem> pathItemsByPath = new LinkedHashMap<>();
+
+    private final OpenAPI openAPI = new OpenAPI();
+
+    private String prefix = "/";
+
+    public String convertToString(ConfigItem item, String prefix, String acceptType) {
+        OpenAPI openapi = convert(item, prefix);
+
+        try {
+            return convertOpenAPIToString(openapi, acceptType);
+        } catch (JsonProcessingException e) {
+            return "";
+        }
+    }
+
+    /**
+     * convert open api object to specific string (default is yaml)
+     *
+     * @param openapi schema
+     * @param acceptType (json / yaml)
+     * @return
+     * @throws JsonProcessingException
+     */
+    public String convertOpenAPIToString(OpenAPI openapi, String acceptType) throws JsonProcessingException {
+        ObjectMapper objectMapper;
+        try {
+            org.springframework.http.MediaType mediaType = org.springframework.http.MediaType.valueOf(acceptType);
+            if (org.springframework.http.MediaType.APPLICATION_JSON.equals(mediaType) && mediaType != null) {
+                objectMapper = new ObjectMapper();
+            } else {
+                objectMapper = new ObjectMapper(new YAMLFactory());
+            }
+        } catch (Exception e) {
+            LOG.warn("UNKNOWN MideaType: " + acceptType + " error: " + e.getMessage());
+            objectMapper = new ObjectMapper(new YAMLFactory());
+        }
+
+        objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        objectMapper.enable(SerializationFeature.INDENT_OUTPUT);
+
+        //TODO: dirty hack to remove exampleSetFlag
+        if (objectMapper.getFactory() instanceof YAMLFactory) {
+            final String intermediateJson = objectMapper.writeValueAsString(openapi);
+            final String almostSwaggerJson = intermediateJson.replaceAll("[\\n\\r\\s]*exampleSetFlag.*,", "");
+            return almostSwaggerJson.replaceAll(",?[\\n\\r\\s]*exampleSetFlag.*", "");
+        } else {
+            final String intermediateJson = objectMapper.writeValueAsString(openapi);
+            final String almostSwaggerJson = intermediateJson.replaceAll("[\\n\\r\\s]*\"exampleSetFlag\".*,", "");
+            return almostSwaggerJson.replaceAll(",?[\\n\\r\\s]*\"exampleSetFlag\".*", "");
+        }
+    }
+
+    public OpenAPI convert(ConfigItem item, String prefix) {
+        this.prefix = prefix;
+
+        // Create an empty set of components
+        Components components = new Components();
+        openAPI.setComponents(components);
+
+        // Create a basic info section
+        // TODO: handle version properly
```

Comment:
please add your name to the TODO so we know later who need to fix what, e.g. TODO: Freddy

Reply:
added